mongo/collection.go

```diff
@@ -68,8 +68,34 @@ func closeImplicitSession(sess *session.Client) {
 	}
 }
 
+// mergeCollectionOptions combines the given CollectionOptions instances into a single *CollectionOptions in a
+// last-one-wins fashion.
```

Comment:
Optional: what do you think of this change? It emphasizes that it's not the last object wins, but rather the last property defined for each object.

```suggestion
// last-property-wins fashion.
```